Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.lean

```diff
@@ -646,6 +649,149 @@ instance : ClosedUnderRestriction (contDiffGroupoid n I) :=
 
 end contDiffGroupoid
 
+section analyticGroupoid
+
+variable {𝕜 : Type _} [NontriviallyNormedField 𝕜] {E : Type _} [NormedAddCommGroup E]
+  [NormedSpace 𝕜 E] {H : Type _} [TopologicalSpace H] (I : ModelWithCorners 𝕜 E H) {M : Type _}
+  [TopologicalSpace M]
+
+/-- Given a model with corners `(E, H)`, we define the groupoid of analytic transformations of `H`
+as the maps that are analytic and map interior to interior when read in `E` through `I`. We also
+explicitly define that they are `C^∞` on the whole domain, since we are only requiring
+analyticity on the interior of the domain. -/
+def analyticGroupoid : StructureGroupoid H :=
+  (contDiffGroupoid ∞ I) ∩ Pregroupoid.groupoid
+    { property := fun f s => AnalyticOn 𝕜 (I ∘ f ∘ I.symm) (I.symm ⁻¹' s ∩ interior (range I)) ∧
+        (I.symm ⁻¹' s ∩ interior (range I)).image (I ∘ f ∘ I.symm) ⊆ interior (range I)
```

Comment:
This is quite a complicated definition; it looks right to me, but I think I'll invite further comment on Zulip.

The following is true, right:
```lean
theorem mem_analyticGroupoid_of_boundaryless [CompleteSpace 𝕜] [I.Boundaryless]
    (e : LocalHomeomorph H H) :
    e ∈ analyticGroupoid I ↔ AnalyticOn 𝕜 (I ∘ e ∘ I.symm) (I '' e.source) := by
  sorry
```
Assuming I've got that right, it would increase my confidence in the definition if we added this.

Reply:
Asked [here](https://github.com/leanprover-community/mathlib4/pull/6386) on Zulip.

Reply:
Will fill in the proof of `mem_analyticGroupoid_of_boundaryless`!

Reply:
@ocfnash Sorry it took a while to follow up on this! Proof should be added now.